File: DataScienceTemp/Statistics/R/EstimatesOfLocation.R
library("matrixStats")

state <- read.csv("../datasets/state.csv")
population_mean <- mean(state[["Population"]])
population_mean_trimm <- mean(state[["Population"]], trim = 0.1)
population_median <- median(state[["Population"]])

message(sprintf("Population mean: %f",population_mean))
message(sprintf("Population mean trim (0.1): %f", population_mean_trimm))
message(sprintf("Population median: %f", population_median))

population_weighted_mean <- weighted.mean(state[["Murder.Rate"]], w=state[["Population"]])
population_weighted_median <- weightedMedian(state[["Murder.Rate"]], w=state[["Population"]])
message(sprintf("Population mean weight: %f", population_weighted_mean))
message(sprintf("Population median weight %f", population_weighted_median))

File: Statistics/Python/EstimatesOfLocation.py
import pandas as pd
import numpy as np
import wquantiles
from scipy.stats import trim_mean

state = pd.read_csv("../datasets/state.csv")

mean = state["Population"].mean()
print("Population mean: {}".format(mean))

trim_mean = trim_mean(state["Population"], 0.1)
print("Population trim mean using 0.1 is: {}".format(trim_mean))

median = state["Population"].median()
print("Population median is: {}".format(median))

weighted_mean = np.average(state["Murder.Rate"], weights=state["Population"])
print("Murder Rate weighted mean based on Population rate: {}".format(weighted_mean))

weighted_median = wquantiles.median(state["Murder.Rate"], weights=state["Population"])
print("Murder Rate weighted median based on Population: {}".format(weighted_median))
